TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/ShooterSystem.java

```diff
@@ -5,18 +5,51 @@
 import com.arcrobotics.ftclib.hardware.motors.MotorEx;
 import com.qualcomm.robotcore.hardware.HardwareMap;
 
+import org.firstinspires.ftc.teamcode.TeleOpMain;
+
+import java.util.function.DoubleSupplier;
+
 public class ShooterSystem extends SubsystemBase {
-    private Motor kuunav;
+    private MotorEx Motor_S;
+    private TeleOpMain teleop;
+    public double shooterPower;
+
+    public ShooterSystem(MotorEx ShooterMotor, TeleOpMain tele){
+        Motor_S = ShooterMotor;
+        teleop = tele;
+    }
 
-    public ShooterSystem(Motor ShooterMotor){
-        kuunav = ShooterMotor;
+    public void adjust(){
+        //These are just place holder values need real ones
+        switch(teleop.pwrSelect) {
+            case 1:
+                shooterPower = 0.5;
+                teleop.telemetry.addData("Shooter Power:", shooterPower);
+                teleop.telemetry.update();
+                break;
+            case 2:
+                shooterPower = 0.75;
+                teleop.telemetry.addData("Shooter Power:", shooterPower);
+                teleop.telemetry.update();
+                break;
+            case 3:
+                shooterPower = 1;
+                teleop.telemetry.addData("Shooter Power:", shooterPower);
+                teleop.telemetry.update();
+                break;
+            default:
+                shooterPower = 0;
+                teleop.telemetry.addData("Shooter Power:", shooterPower);
+                teleop.telemetry.update();
+                break;
+        }
```

Comment:
This isn't going to work. Pass a `DoubleSupplier` into the constructor. That `DoubleSupplier` will be a reference to the `pwrSelect` value. In your TeleOp, do something like:
```java
// dpad buttons declaration - outside the method
private Button dpadDown, dpadUp;
// create your pwrSelect value - name this whatever you want
public double pwrSelect;  // I'm making this public just in case of something wacky

/* in your initialize() method */
dpadDown = new GamepadButton(m_toolOp, GamepadKeys.Button.DPAD_DOWN)
                .whenPressed(new InstantCommand(() -> {
                    if (pwrSelect < 0.05) {
                        pwrSelect = 1;
                    } else {
                        pwrSelect -= 0.25;
                    }
                }));
dpadUp = new GamepadButton(m_toolOp, GamepadKeys.Button.DPAD_UP)
                .whenPressed(new InstantCommand(() -> {
                    if (pwrSelect > 0.95) {
                        pwrSelect = 0;
                    } else {
                        pwrSelect += 0.25;
                    }
                }));

// your new ShooterSystem mechanic
// your constructor should be ShooterSystem(MotorEx, Telemetry, DoubleSupplier)
shooterSystem = new ShooterSystem(shot, telemetry, () -> pwrSelect);
```